### DISCUSSION TITLE
MvxPreferenceSetupHelper in 7.1

### DISCUSSION THREAD

**Discussion by DavidMarquezF** [MvvmCross/MvvmCross#4091] [Q&A] [answered]
I'm not sure if this should be considered an issue or not, so that's why I'm first asking here. Prior to the whole migration to AndroidX you were able to bind the preference fragment view in the followind manner:

```
set.Bind(updateOptions)
                    .For(v => v.BindClick())
                    .To(vm => vm.UpdateOptionsCommand);
```

To do so I believe you first had to register MvxPreferenceSetupHelper:

```
MvxPreferenceSetupHelper.FillTargetFactories(registry)
```

Now Preference does not contain a definition for BindClick and MvxPreferenceSetupHelper has been removed (or at least I haven't been able to find it) 
What happened to this now? Is it a desired change? Or it's just somethng that needs to be done?

**Comment by Cheesebaron** [comment, marked as answer]
AndroidX PreferenceFragment is here: https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross/Platforms/Android/Views/Fragments/MvxPreferenceFragment.cs

You might be right though. Seems like I've accidentally omitted some target bindings when I did the AndroidX migration.

Workaround right now would be
1. grab the classes in: https://github.com/MvvmCross/MvvmCross/tree/6.4.2/MvvmCross.Android.Support/V7.Preference/Target 
2. migrate them to AndroidX types
3. grab https://github.com/MvvmCross/MvvmCross/blob/6.4.2/MvvmCross.Android.Support/V7.Preference/MvxPreferencePropertyBinding.cs and https://github.com/MvvmCross/MvvmCross/blob/6.4.2/MvvmCross.Android.Support/V7.Preference/MvxPreferencePropertyBindingExtensions.cs and https://github.com/MvvmCross/MvvmCross/blob/6.4.2/MvvmCross.Android.Support/V7.Preference/MvxPreferenceSetupHelper.cs and add them back somewhere in your project. These files probably need some adjustments too.

I'll create an issue for this.

**Reply by Cheesebaron** [reply]
Tracked in: https://github.com/MvvmCross/MvvmCross/issues/4094
